app/models/subnet.rb

```diff
@@ -152,12 +152,16 @@ def cidr=(cidr)
     nil
   end
 
-  def supports_ipam_mode?(mode)
-    self.class.supported_ipam_modes.include?(mode)
+  def self.supports_ipam_mode?(mode)
+    supported_ipam_modes.include?(mode)
+  end
+
+  def self.supports_dhcp?
```

Comment:
I think you can do without this method and use `supports_ipam_mode?(:dhcp)` easily. Not only that, but also you can avoid calling `self.class` so often by using delegate.

```ruby
delegate :supports_ipam_mode?, to: 'self.class'
```

That would allow you to call `supports_ipam_mode?(whatever)` everywhere in a cleaner way (in my opinion)